pkg/applicationserver/io/packages/loragls/v3/package.go

```diff
@@ -87,6 +109,22 @@ func (p *GeolocationPackage) Package() *ttnpb.ApplicationPackage {
 	}
 }
 
+// assocIDs returns the identifiers of the given association. If the association is nil, new identifiers are created.
+func (p *GeolocationPackage) assocIDs(
+	assoc *ttnpb.ApplicationPackageAssociation,
+	ids *ttnpb.EndDeviceIdentifiers,
+) *ttnpb.ApplicationPackageAssociationIdentifiers {
+	assocIDs := assoc.GetIds()
+	if assocIDs == nil {
+		pkg := p.Package()
```

Comment:
You should use the port of the default association, and then you no longer need this function to be tied to `*GeolocationPackage`.

Due to checks in the uplink handler, you will always have at least the association, or the default association, or both.